src/Cli/Exporter.cs

```diff
@@ -115,27 +118,66 @@ private static async Task ExportGraphQL(ExportOptions options, RuntimeConfig run
+        /// Fetches the GraphQL schema from the DAB service, attempting to use the HTTPS endpoint first.
+        /// If the HTTPS endpoint fails, it falls back to using the HTTP endpoint.
+        /// Logs the process of fetching the schema and any fallback actions taken.
+        /// </summary>
+        /// <param name="runtimeConfig">The runtime config object containing the GraphQL path.</param>
+        /// <param name="logger">The logger instance used to log information and errors during the schema fetching process.</param>
+        /// <returns>The GraphQL schema as a string.</returns>
+        internal string ExportGraphQLFromDabService(RuntimeConfig runtimeConfig, ILogger logger)
         {
             string schemaText;
             // Fetch the schema from the GraphQL API
             logger.LogInformation("Fetching schema from GraphQL API.");
 
-            HttpClient client = new( // CodeQL[SM02185] Loading internal server connection
+            try
+            {
+                logger.LogInformation("Trying to fetch schema from DAB Service using HTTPS endpoint.");
+                schemaText = GetGraphQLSchema(runtimeConfig, useFallbackURL: false);
+            }
+            catch
+            {
+                logger.LogInformation("Failed to fetch schema from DAB Service using HTTPS endpoint. Trying with HTTP endpoint.");
+                schemaText = GetGraphQLSchema(runtimeConfig, useFallbackURL: true);
+            }
+
+            return schemaText;
+        }
+
+        /// <summary>
+        /// Retrieves the GraphQL schema from the DAB service using either the HTTPS or HTTP endpoint based on the specified fallback option.
+        /// </summary>
+        /// <param name="runtimeConfig">The runtime configuration containing the GraphQL path and other settings.</param>
+        /// <param name="useFallbackURL">A boolean flag indicating whether to use the fallback HTTP endpoint. If false, the method attempts to use the HTTPS endpoint.</param>
+        internal virtual string GetGraphQLSchema(RuntimeConfig runtimeConfig, bool useFallbackURL = false)
+        {
+            HttpClient client;
+            if (!useFallbackURL)
+            {
+                client = new( // CodeQL[SM02185] Loading internal server connection
                                                     new HttpClientHandler { ServerCertificateCustomValidationCallback = HttpClientHandler.DangerousAcceptAnyServerCertificateValidator }
                                                 )
+                {
+                    BaseAddress = new Uri($"https://localhost:5001{runtimeConfig.GraphQLPath}")
+                };
+            }
+            else
             {
-                BaseAddress = new Uri($"https://localhost:5001{runtimeConfig.GraphQLPath}")
-            };
+                client = new()
+                {
+                    BaseAddress = new Uri($"http://localhost:5000{runtimeConfig.GraphQLPath}")
+                };
+            }
```

Comment:
We need to rethink this function now that we're visiting it in your PR. Few thoughts:
1. we hardcode localhost:5001/5000 -> real use cases may override these ports and endpoints.  I don't recall whether we restrict GraphQL schema export for dev mode only. (i don't think we have restrictions here).
2. we use `ServerCertificateCustomValidationCallback = HttpClientHandler.DangerousAcceptAnyServerCertificateValidator` which is insecure for production. I think this delegate should only be set when developer does not use my other discussion's proposed `--http-fallback` or `--allow-insecure-fallback`.
3. order of operations:( let's discuss this)
    1. HTTPS (valid certificate, not self-signed). -> `dab export graphql`
    2. HTTPS (self-signed dev scenario, only works with fallback flag, set the `HttpClientHandler.DangerousAcceptAnyServerCertificateValidator` delegate
    3. If step 2 fails, then fall back again to http

Reply:
the previous code says usage of `DangerousAcceptAnyServerCertificateValidator` was exempt here because of internal server usage.

I dont think we have an API that exposes the export functionality. The only usage seems to be via the CLI from the export command for a short duration so unlikely to have production mode usage. Hence, wonder if we need to add additional flags for such a simple functionality?

Reply:
yes, export functionality is only for CLI.

Reply:
I think 1 is still important here even if you decide to continue usage of `HttpClientHandler.DangerousAcceptAnyServerCertificateValidator`. We hard code the port used to host dab and don't consider a developer overriding those ports which could be possible for development mode.

Reply:
please open an issue to track customizing the port used for export. port -> ports 5000/5001 are hardcoded and should allow customization through option to CLI. This thread can be closed using the issue number created.